lib/HLSL/DxilCondenseResources.cpp

```diff
@@ -567,7 +567,7 @@ class DxilLowerCreateHandleForLib : public ModulePass {
     bChanged |= RemovePhiOnResource();
 
     if (m_bLegalizationFailed)
-      return bChanged;
+      throw hlsl::Exception(DXC_E_OPTIMIZATION_FAILED, "Resource legalization failed");
```

Comment:
I understand that this was used to test the change, but let's not change the behavior of this pass just yet.  If we do want to abort compilation on an error in this pass, there is more work to do to capture more failure cases that `m_bLegalizationFailed` doesn't capture.  We can save that for a different PR.

Reply:
This change is also causing the test failure, so let's back it out for now.